server/src/session.ts

```diff
@@ -850,6 +855,17 @@ export class Session {
     } else {
       // Immediately enable Legacy/ViewEngine language service
       this.info(`Enabling VE language service for ${projectName}.`);
+      if (!this.angularCoreVersion.has(project)) {
+        const angularCore = resolveAngularCore(this.ngProbeLocations);
```

Comment:
`ngProbeLocations` is a collection of paths which we would like to resolve `@angular/language-service` from, and it is not the same as the location of `@angular/core` for the project. For example, the probe locations include `ngdk` and the extension directory where `@angular/language-service` is bundled.
Here, we should resolve `@angular/core` from 
```suggestion 
        const angularCore = resolve(
          '@angular/core', 
          project.getFileNames().find(f => f.endsWith('@angular/core/core.d.ts')!
        );
```
The `@angular/core/core.d.ts` file is guaranteed to exist because it's checked here:
https://github.com/angular/vscode-ng-language-service/blob/0abe4094f4a9af45c53dcabb7e4cf470dedcc50e/server/src/session.ts#L864-L865
We should just refactor `checkIsAngularProject()` to return the file name if the project is Angular.
https://github.com/angular/vscode-ng-language-service/blob/0abe4094f4a9af45c53dcabb7e4cf470dedcc50e/server/src/session.ts#L856-L860

Reply:
Done.